Client/src/Pages/Status/CreateStatus/index.jsx

```diff
@@ -0,0 +1,97 @@
+import PropTypes from "prop-types";
+import { useNavigate } from "react-router";
+import { useEffect, useState } from "react";
+import { Box, Tab, useTheme } from "@mui/material";
+import TabContext from "@mui/lab/TabContext";
+import TabList from "@mui/lab/TabList";
+import GeneralSettingsPanel from "../../../Components/TabPanels/Status/GeneralSettingsPanel";
+import {
+	capitalizeFirstLetter,
+	toLowerCaseFirstLetter,
+} from "../../../Utils/stringUtils";
+import ContentPanel from "../../../Components/TabPanels/Status/ContentPanel";
+
+/**
+ * CreateStatus page renders a page with tabs for general settings and contents.
+ * @param {string} [props.open] - Specifies the initially open tab: 'general settings' or 'content'.
+ * @returns {JSX.Element}
+ */
+
+const CreateStatus = ({ open = "general-settings" }) => {
+	const theme = useTheme();
+	const navigate = useNavigate();
+	let tabList = ["General Settings", "Contents"];
+	const tab = open
+		.split("-")
+		.map((a) => capitalizeFirstLetter(a))
+		.join(" ");
+
+	const [tabIdx, setTabIdx] = useState(tabList.indexOf(tab));
+
+	const handleTabChange = (event, newTab) => {
+		let toNavString = newTab.split(" ").map((a) => toLowerCaseFirstLetter(a));
+		setTabIdx(tabList.indexOf(newTab));
+		toNavString = toNavString.join("-");
+		navigate(`/status/${toNavString}`);
+	};
+
+	useEffect(() => {
+		setTabIdx(tabList.indexOf(tab));
+	}, [tab]);
```

Comment:
if we define the tab state on the url we probably won't need the effect

Reply:
this is to show the correspondent tabpanel when switched tab

Reply:
Not if we get our state from the url. Call me if you want me to clarify that

Reply:
in theory that is correct, however event.target.name is empty in this case, I think we should be open to new way , in this case, tablist provided out of box the onChange that gave us the tab value already, why not just use that?

Reply:
Here we are discussing something else, state on the url. The onChange x nae would be on the other thread

Reply:
@marcelluscaio would you mind showing a quick implementation of what you mean here?  I think I understand what you're suggesting but it would be good to see it in on paper so we're all on the same page